DQMOffline/Muon/src/MuonTiming.cc

```diff
@@ -0,0 +1,196 @@
+    yields[iObjectName]->setBinLabel(7, "Combined not RPC");
+    yields[iObjectName]->setBinLabel(8, "RPC not Combined");
+    yields[iObjectName]->setBinLabel(9, "Not passing veto");
+    yields[iObjectName]->setBinLabel(10, "Passing veto");
+    etaptVeto[iObjectName]->setAxisTitle("p_{T} [GeV]");
+    etaptVeto[iObjectName]->setAxisTitle("#eta#", 2);
+    ptVeto[iObjectName]->setAxisTitle("p_{T} [GeV]");
+    etaVeto[iObjectName]->setAxisTitle("#eta");
+  }
+
+}
+
+
+
+void MuonTiming::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup){
+
+  LogTrace(metname)<<"[MuonTiming] Analyze the mu";
+  theService->update(iSetup);
+ 
+  // Take the muon container
+  edm::Handle<edm::View<reco::Muon> > muons; 
+  iEvent.getByToken(theMuonCollectionLabel_,muons);
+ 
+
+  if(!muons.isValid()) return;
+
+  for (edm::View<reco::Muon>::const_iterator muon = muons->begin(); muon != muons->end(); ++muon){
+    const reco::MuonTime time = muon->time(); 
+    const reco::MuonTime rpcTime = muon->rpcTime();
+    //Only creating so far the timing information for STA muons
+    if(!muon->isStandAloneMuon() || muon->isGlobalMuon()) continue;
+    reco::TrackRef track;
+    //Select whether it's a global or standalone muon
+    object theObject = sta;
+    if(muon->isGlobalMuon()) {track = muon->combinedMuon(); theObject = glb;}
+    else {track = muon->standAloneMuon(); theObject = sta;}
+
+    //These definitions have been taken from Piotr Traczyk  
+    bool cmbok =(time.nDof > 7);
+    bool rpcok =(rpcTime.nDof > 1 && rpcTime.timeAtIpInOutErr == 0);
+    bool veto = false;
+    if (rpcok) {
+       if ((fabs(rpcTime.timeAtIpInOut) > 10) && !(cmbok && fabs(time.timeAtIpInOut)<10)) veto=true;
+       else if (cmbok && (time.timeAtIpInOut > 20 || time.timeAtIpInOut<-45)) veto=true;
+    } 
+
+    //std::cout << time.timeAtIpInOut << std::endl;    
+    //Filling the yields histogram
+    if(muon->isTimeValid()) yields[theObject]->Fill(1);
+    else yields[theObject]->Fill(0);
+  
+    if(cmbok) yields[theObject]->Fill(3);
+    else yields[theObject]->Fill(2);
+    
+    if(rpcok) yields[theObject]->Fill(5);
+    else yields[theObject]->Fill(4);
+    
+    if(cmbok && !rpcok) yields[theObject]->Fill(6);
+    if(!cmbok && rpcok) yields[theObject]->Fill(7);
+    
+    if(veto) yields[theObject]->Fill(9);
+    else yields[theObject]->Fill(9);
+
+    //Starting now with the pt and eta for vetoed and not vetoed muons
+    if(veto) {
+      etaptVeto[theObject]->Fill(track->pt(), track->eta());
+      etaVeto[theObject]->Fill(track->eta());
+      ptVeto[theObject]->Fill(track->pt());
+    } 
+    
+    //Check the eta region of the muon
+    eta theEta = barrel;
+    if(fabs(track->eta()) > etaBarrelMin && fabs(track->eta()) < etaBarrelMax) theEta = barrel;
+    if(fabs(track->eta()) > etaOverlapMin && fabs(track->eta()) < etaOverlapMax) theEta = overlap;
+    if(fabs(track->eta()) > etaEndcapMin && fabs(track->eta()) < etaEndcapMax) theEta = endcap;
```

Comment:
with this logic and your parameters, theEta will never = overlap.
otherwise, please also consider a >= to correctly handle bin boundaries